sklearn/preprocessing/data.py

```diff
@@ -1901,3 +1909,458 @@ def transform(self, X):
+        If RandomState instance, random_state is the random number generator;
+        If None, the random number generator is the RandomState instance used
+        by np.random.
+
+    Attributes
+    ----------
+    quantiles_ : ndarray, shape (n_quantiles, n_features)
+        The values corresponding the quantiles of reference.
+
+    See also
+    --------
+    :class:`sklearn.preprocessing.StandardScaler` to perform standardization
+    that is faster, but less robust to outliers.
+
+    :class:`sklearn.preprocessing.RobustScaler` to perform robust
+    standardization that removes the influence of outliers but does not put
+    outliers and inliers on the same scale.
+    """
+
+    def __init__(self, n_quantiles=1000, subsample=int(1e5),
+                 ignore_implicit_zeros=False, output_pdf='uniform',
+                 random_state=None):
+        self.n_quantiles = n_quantiles
+        self.subsample = subsample
+        self.ignore_implicit_zeros = ignore_implicit_zeros
+        self.output_pdf = output_pdf
+        self.random_state = random_state
+
+    def _build_f(self):
+        """Build the transform functions."""
+        check_is_fitted(self, 'quantiles_')
+
+        if self.ignore_implicit_zeros:
+            warnings.warn("'ignore_implicit_zeros' takes effect only with"
+                          " sparse matrix. This parameter has no effect.")
+
+        references = np.linspace(0, 1, self.n_quantiles, endpoint=True)
+
+        self._f_transform = tuple([
+            interp1d(quantiles_feature, references,
+                     bounds_error=False,
+                     fill_value=0.)
+            for quantiles_feature in self.quantiles_.T])
+
+        self._f_inverse_transform = tuple([
+            interp1d(references, quantiles_feature,
+                     bounds_error=False,
+                     fill_value=0.)
+            for quantiles_feature in self.quantiles_.T])
+
+    def _dense_fit(self, X):
+        """Compute percentiles for dense matrices.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_samples, n_features)
+            The data used to scale along the features axis.
+        """
+        rng = check_random_state(self.random_state)
+
+        # subsample the matrix X if necessary
+        n_samples, n_features = X.shape
+        if self.subsample < n_samples:
+            subsample_idx = rng.permutation(range(n_samples))[:self.subsample]
+        else:
+            subsample_idx = range(n_samples)
+
+        # for compatibility issue with numpy<=1.8.X, references
+        # need to be a list scaled between 0 and 100
+        references = np.linspace(0, 1, self.n_quantiles,
+                                 endpoint=True).tolist()
+        # references is a list that we need to scale between
+        # 0 and 100.
+        self.quantiles_ = np.array([np.percentile(
+            X[subsample_idx, feature_idx], [x * 100 for x in references])
```

Comment:
why do we do the multiplication here not in `linspace` or vectorized over the array above?